<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro" name="AGV" >
<xacro:property name="room_length"  value="$(arg length)" />
<xacro:property name="room_width"  value="$(arg width)" />

    <link name="wall_1">
        <visual>
            <geometry>
                <box size="${room_length+0.2} 0.2 3"/>
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <material name="grey">
                <color rgba="0.8 0.9 1 1"/>
            </material>
        </visual>
    </link>

    <joint name="wall_joint_1" type="fixed">
	<parent link="wall_1"/>
	<child link="wall_2"/>
    </joint>

    <link name="wall_2">
        <visual>
            <geometry>
                <box size="${room_width+0.2} 0.2 3"/>
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <material name="grey">
                <color rgba="0.8 0.9 1 1"/>
            </material>
        </visual>
    </link>

    <joint name="wall_joint_2" type="fixed">
	<parent link="wall_2"/>
	<child link="wall_3"/>
    </joint>

    <link name="wall_3">
        <visual>
            <geometry>
                <box size="${room_length+0.2} 0.2 3"/>
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <material name="grey">
                <color rgba="0.8 0.9 1 1"/>
            </material>
        </visual>
    </link>


</robot>
